src/Serializer/UserSerializerHandler.php

```diff
@@ -13,18 +13,81 @@
 
 namespace Sonata\UserBundle\Serializer;
 
-use Sonata\CoreBundle\Serializer\BaseSerializerHandler;
+use JMS\Serializer\Context;
+use JMS\Serializer\GraphNavigatorInterface;
+use JMS\Serializer\Handler\SubscribingHandlerInterface;
+use JMS\Serializer\Visitor\SerializationVisitorInterface;
+use Sonata\Doctrine\Model\ManagerInterface;
 
 /**
  * @author Sylvain Deloux <sylvain.deloux@ekino.com>
  */
-class UserSerializerHandler extends BaseSerializerHandler
+final class UserSerializerHandler implements SubscribingHandlerInterface
 {
     /**
-     * {@inheritdoc}
+     * @var ManagerInterface
      */
-    public static function getType()
+    private $manager;
+
+    /**
+     * @var string[]
+     */
+    private static $formats = ['json', 'xml', 'yml'];
+
+    public function __construct(ManagerInterface $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    public static function getSubscribingMethods(): array
+    {
+        $type = 'sonata_user_user_id';
+        $methods = [];
+
+        foreach (static::$formats as $format) {
+            $methods[] = [
+                'direction' => GraphNavigatorInterface::DIRECTION_SERIALIZATION,
+                'format' => $format,
+                'type' => $type,
+                'method' => 'serializeObjectToId',
+            ];
+
+            $methods[] = [
+                'direction' => GraphNavigatorInterface::DIRECTION_DESERIALIZATION,
+                'format' => $format,
+                'type' => $type,
+                'method' => 'deserializeObjectFromId',
+            ];
+        }
+
+        return $methods;
+    }
+
+    /**
+     * Serialize data object to id.
+     *
+     * @param object $data
+     *
+     * @return int|null
```

Comment:
Couldn't the object use a string for its id? (like UUID by instance).

Reply:
That's why I thought if this class is used or not. I haven't found any reference to `serialized-name` or `'sonata_user_user_id'` like the [doc says](https://github.com/sonata-project/SonataCoreBundle/pull/41/files#diff-4a1e47a8a97d01d2f85664bbce2ddf2dR24).